containers/ecr-viewer/src/app/admin/user/UserForm.tsx

```diff
@@ -0,0 +1,265 @@
+
+import {
+  Button,
+  RequiredMarker,
+  TextInput,
+  Radio,
+} from "@trussworks/react-uswds";
+
+import { FieldSet } from "@/app/components/forms/FieldSet";
+import { FormPageContent } from "@/app/components/forms/FormPageContent";
+import { ToastContext } from "@/app/components/toast/ToastProvider";
+import { ServerActionResult } from "@/app/services/errorService";
+import { ListedProgramArea } from "@/app/services/programAreaService";
+import { makePlural, toKebabCase, toTitleCase } from "@/app/utils/format-utils";
+import { ExpandCollapseAccordionControlled } from "@/app/view-data/components/ExpandCollapseAccordion";
+import { AccordionItem } from "@/app/view-data/types";
+
+type UserType = "admin" | "standard";
+
+interface FormProgram extends ListedProgramArea {
+  checked?: boolean;
+}
+
+interface FormValues {
+  email?: string;
+  userType?: UserType;
+  programs: FormProgram[];
+}
+
+/**
+ *
+ * @param props React props
+ * @param props.initValues Initial values the form is set to
+ * @param props.action Action of the form (e.g. "Create", "Edit")
+ * @param props.submitAction Handler for the submitted data
+ * @returns Program area add/edit form
+ */
+export const UserForm = ({
+  action,
+  initValues,
+  submitAction,
+}: {
+  action: string;
+  initValues: FormValues;
+  submitAction: (
+    email: string,
+    userType: UserType,
+    programs: string[],
+  ) => Promise<ServerActionResult<string>>;
+}) => {
+  const [email, setEmail] = useState(initValues.email || "");
+  const [userType, setUserType] = useState<UserType>("standard");
+  const [programs, setPrograms] = useState(initValues.programs);
+
+  const { createToast } = React.useContext(ToastContext);
+
+  const selectedPrograms = Object.values(programs)
+    .filter(({ checked }) => !!checked)
+    .map(({ uuid }) => uuid);
+  const numProgramsSelected = selectedPrograms.length;
+
+  const valid = !!email && numProgramsSelected > 0;
+
+  return (
+    <FormPageContent
+      action={`${action} user`}
+      formValid={valid}
+      submitAction={async () => {
+        const res = await submitAction(
+          email.trim(),
+          userType,
+          selectedPrograms,
+        );
+        if (!res.error)
+          createToast(`${email.trim()} successfully saved`, "success");
+        return res;
+      }}
+      successRoute="/admin/user"
+    >
+      <EmailFieldSet email={email} setEmail={setEmail} />
+      <UserTypeFieldSet userType={userType} setUserType={setUserType} />
+      <ProgramFieldSet
+        programs={programs}
+        setPrograms={setPrograms}
+        numProgramsSelected={numProgramsSelected}
+      />
+    </FormPageContent>
+  );
+};
+
+const EmailFieldSet = ({
+  email,
+  setEmail,
+}: {
+  email: string;
+  setEmail: (n: string) => void;
+}) => {
+  return (
+    <FieldSet legend="Email">
+      <span>
+        Add the new user by their login email (<RequiredMarker />)
+      </span>
+      <label className="usa-label">
+        EMAIL
```

Comment:
@ashton-skylight in the program form the labels are sentence case, but here it's all caps - which should it be?